src/middlewares/authMiddleware.ts

```diff
@@ -0,0 +1,44 @@
+import { Request, Response, NextFunction } from 'express';
+import jwt from 'jsonwebtoken';
+
+/**
+ * Middleware para autenticar usuários com base em um token JWT.
+ * Verifica se o token é válido e o decodifica, adicionando os dados do
+ * usuário ao req.user. Se o token for inválido, retorna um erro 403.
+ * @param req - Objeto da requisi o HTTP.
+ * @param res - Objeto da resposta HTTP.
+ * @param next - Função de callback para passar o controle para o próximo middleware.
+ */
+export async function authenticateJWT(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const token = req.headers.authorization?.split(' ')[1];
+  const SECRET_KEY = process.env.JWT_SECRET || 'ChaveSecreta';
```

Comment:
Dúvida: temos um arquivo env.example? Seria legal manter mapeado as envs que temos pra quando alguém for rodar, ter um norte do que precisa levantar pra subir a API, ai só precisa ter as chaves  e os valores qualquer coisa com uma explicação breve do que é

Reply:
irei adicionar